src/servant/tt_000/10E8.c

```diff
@@ -717,19 +706,19 @@ void func_80172120(Entity* self) {
             g_api.func_8011A3AC(self, 15, 1, &D_80174C30);
             self->ext.bat.unk86 = 0xC00;
             func_801710E8(self, D_801705EC);
-            func_80171568(self);
+            CreateBlueTrailEntity(self);
             self->step++;
         }
         break;
     case 3:
         D_80174B1C = self->ext.bat.target->posX.i.hi;
         D_80174B20 = self->ext.bat.target->posY.i.hi;
-        D_80174B0C = func_80173F30(self, (s16)D_80174B1C, (s16)D_80174B20);
+        D_80174B0C = func_80173F30(self, D_80174B1C, D_80174B20);
         D_80174B10 = func_80173F74(D_80174B0C, self->ext.bat.unk86,
-                                   (s16)D_80170658[D_80174C30.level / 10][1]);
+                                   D_80170658[D_80174C30.level / 10][1]);
         self->ext.bat.unk86 = D_80174B10;
-        self->velocityX = rcos(D_80174B10) << 6;
-        self->velocityY = -(rsin(D_80174B10) << 6);
+        self->velocityX = rcos(D_80174B10) << 2 << 4;
```

Comment:
I wonder if this `<< 2 << 4` is a clue on how the developers used to use rsin/rcos values. Maybe they had a macro or something